### PR TITLE
US-7 (FIX): skip null the error + some refactor + fixes

### PR DESCRIPTION
**Pull request by mackwic**
Changes: 
- skip null the error
- introduce `onError` event for testing purpose
- refactor tests to be unit tests (rather than integration)
- introduce `done` callbacks when necessary in integration testing

### CODE REVIEW

**Comment by twickham** [comment]
I've deployed this PR to http://7-skip-une-reponse.pix.beta.gouv.fr. Please check it out

**Comment by mackwic** [comment]
Yay \o/

File: app/components/challenge-item.js [hunk 1]
```diff
@@ -33,13 +33,17 @@ const ChallengeItem = Ember.Component.extend({
   actions: {
     validate(challenge, assessment) {
       if (Ember.isEmpty(this.get('selectedProposal'))) {
-        this.set('error', 'Vous devez sélectionner une réponse.');
+
+        const errorMsg = 'Vous devez sélectionner une réponse.';
```
Comment (jbuget) [inline app/components/challenge-item.js]: rename variable "errorMsg" to "errorMessage"

**Comment by jbuget** [comment]
Hormis le renommage de variable, OK pour moi

**Comment by twickham** [comment]
I've deployed this PR to http://7-skip-une-reponse.pix.beta.gouv.fr. Please check it out

**Pull request merged**